### DISCUSSION TITLE
Node not found

### DISCUSSION THREAD

**Discussion by alecfilios** [gstavrinos/ai_msc_robotic_systems_assignments#6] [Q&A] [answered]
Sometimes after a change in the code has taken place, the following error occurs: 
[ros2-29] Node not found
[ros2-34] Unknown transition requested, available ones are:
[ros2-34] - configure [1]
[ros2-34] - shutdown [5]
ROR] [ros2-34]: process has died [pid 25937, exit code 1, cmd 'ros2 lifecycle set table_explorer_node activate'].

A workaround is to colcon build and source again, but that does not always work and a restart of the Ubuntu VM is required.

**Comment by gstavrinos** [comment, marked as answer]
Is your node running when this error occurs? It looks like it is either not running (at the time of the lifecycle call) or there is a problem with your node's lifecycle state handling.
